doc/rst/usingchapel/editors/emacs.rst

```diff
@@ -0,0 +1,69 @@
+.. _emacs:
+
+Emacs
+-----
+
+.. _emacs-syntax:
+
+Syntax Highlighting
+~~~~~~~~~~~~~~~~~~~
+
+To get syntax highlighting for Chapel in Emacs, you can use the files in the top-level ``highlight/emacs`` directory of the Chapel source tree.  These files are licensed under the GPL, see ``highlight/emacs/LICENSE``.  To use them, take the following steps:
+
+1. Copy/move the ``highlight/emacs`` directory to your ``CHPL_HOME`` directory. If you installed Chapel from a package manager, you may need to download the Chapel source code to get these files. To query the correct value of ``CHPL_HOME``, run ``chpl --print-chpl-home``.
+
+2. Add the following lines to your standard ``.emacs`` or ``.emacs.d/init.el`` file:
+
+.. code-block:: emacs-lisp
+
+    ; query the environment variable CHPL_HOME
+    (setq chpl-home (getenv "CHPL_HOME"))
+
+    ; prepend $CHPL_HOME/highlight/emacs to your emacs load-path
+    (if (string= chpl-home nil)
+        nil
+      (add-to-list 'load-path (concat chpl-home
+                                      "/highlight/emacs/"
+                                      )
+                   )
+      )
+
+    ; make sure that when chpl-mode is entered, (our modified) cc-mode is
+    ; loaded
+    (autoload 'chpl-mode "chpl-mode" "Chpl enhanced cc-mode" t)
+
+    ; make loading files with a .chpl extension put emacs into chpl-mode
+    (add-to-list 'auto-mode-alist '("\\.chpl$" . chpl-mode))
+
+
+.. _emacs-lsp:
+
+Language Server Support
+~~~~~~~~~~~~~~~~~~~~~~~
+
+With Emacs 29.1, support has been added for language server protocols via `Eglot
+<https://www.gnu.org/software/emacs/manual/html_mono/eglot.html>`_
+
+To utilize the :ref:`Chapel language server <readme-chpl-language-server>` and/or :ref:`readme-chplcheck` with Eglot, add the following to your
```

Comment:
Something tells me this paragraph is not following the 80 characters rule :) It also looks like you violate it on line 11 and 13